media-libs/libjxl/libjxl-0.3.7.recipe

```diff
@@ -0,0 +1,142 @@
+SOURCE_FILENAME_2="lodepng-$srcGitRev_2.tar.gz"
+srcGitRev_3="64374756e03700d649f897dbd98c95e78c30c7da"
+SOURCE_URI_3="https://skia.googlesource.com/skcms/+archive/$srcGitRev_3.tar.gz"
+CHECKSUM_SHA256_3="3de8601ccaf885d98b48a019ab1c963c6058bc8f819c7d7ee258d06c2d870066"
+SOURCE_FILENAME_3="ckcms-$srcGitRev_3.tar.gz"
+srcGitRev_4="868ab558fad70fcbe8863ba4e85179eeb81cc840"
+SOURCE_URI_4="https://github.com/webmproject/sjpeg/archive/$srcGitRev_4.tar.gz"
+CHECKSUM_SHA256_4="72279cd6d4089b62a49cb127353bf875cb35844eda42d90901dd32f08992060e"
+SOURCE_FILENAME_4="sjpeg-$srcGitRev_4.tar.gz"
+PATCHES="libjxl-$portVersion.patchset"
+
+SOURCE_DIR="jpeg-xl-v$portVersion"
+
+ARCHITECTURES="!x86_gcc2 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	libjxl$secondaryArchSuffix = $portVersion
+	lib:libjxl$secondaryArchSuffix = $portVersion
+	lib:libjxl_dec$secondaryArchSuffix = $portVersion
+	lib:libjxl_threads$secondaryArchSuffix = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libavif$secondaryArchSuffix
+	lib:libbrotlicommon$secondaryArchSuffix
+	lib:libbrotlidec$secondaryArchSuffix
+	lib:libbrotlienc$secondaryArchSuffix
+	lib:libgif$secondaryArchSuffix
+	lib:libGL$secondaryArchSuffix
+	lib:libIex_3_0$secondaryArchSuffix
+	lib:libIlmThread_3_0$secondaryArchSuffix
+	lib:libImath_3_0$secondaryArchSuffix
+	lib:libOpenEXR_3_0$secondaryArchSuffix
+	lib:libOpenEXRUtil_3_0$secondaryArchSuffix
+	lib:libjpeg$secondaryArchSuffix
+	lib:libpng16$secondaryArchSuffix
+	lib:libwebp$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	libjxl${secondaryArchSuffix}_devel = $portVersion
+	devel:libjxl$secondaryArchSuffix = $portVersion
+	devel:libjxl_dec$secondaryArchSuffix = $portVersion
+	devel:libjxl_threads$secondaryArchSuffix = $portVersion
+	"
+REQUIRES_devel="
+	libjxl$secondaryArchSuffix == $portVersion base
+	"
+
+PROVIDES_tools="
+	libjxl${secondaryArchSuffix}_tools = $portVersion
+	cmd:benchmark_xl$secondaryArchSuffix
+	cmd:cjxl$secondaryArchSuffix
+	cmd:djxl$secondaryArchSuffix
+	"
+REQUIRES_tools="
+	$REQUIRES
+	libjxl$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libavif$secondaryArchSuffix
+	devel:libbrotlicommon$secondaryArchSuffix
+	devel:libbrotlidec$secondaryArchSuffix
+	devel:libbrotlienc$secondaryArchSuffix
+	devel:libgif$secondaryArchSuffix
+	devel:libGL$secondaryArchSuffix
+	devel:libhwy$secondaryArchSuffix
+	devel:libhwy_contrib$secondaryArchSuffix
+	devel:libIex_3_0$secondaryArchSuffix
+	devel:libIlmThread_3_0$secondaryArchSuffix
+	devel:libImath_3_0$secondaryArchSuffix
+	devel:libOpenEXR_3_0$secondaryArchSuffix
+	devel:libOpenEXRUtil_3_0$secondaryArchSuffix
+	devel:libjpeg$secondaryArchSuffix
+	devel:libpng16$secondaryArchSuffix
+	devel:libwebp$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:cmake
+	cmd:gcc$secondaryArchSuffix
+	cmd:git
```

Comment:
cmd:git should be removed.

Reply:
If I get around to it I'll have another look at this this week(end) :)

Reply:
Re-working the recipe so it will download the submodules, for this `git` is needed, haikuporter doesn't play nice so I have to delete the `.git` directory so it fetches the `submodules` ...

Reply:
There seems to be an issue (maybe related to this) at haikuporter at: https://github.com/haikuports/haikuporter/issues/62

Reply:
I see Homebrew manages to build without git.
https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/jpeg-xl.rb

Reply:
Others also do it without git, but they don't use the checksum on those or use "SKIP" to by-pass it, the offending checksum fail is especially for skcms (googlesource), I could leave that checksum out and go back to the previous version, that would work without git.

Reply:
I was talking about submodules